### ISSUE TITLE
New community website

### ISSUE THREAD

**Issue by pmatseykanets** [beanstalkd/beanstalkd#395]
This is related to #392 but goes beyond just moving and / or setting a redirect to an existing version of content that hasn't changed for years.

I hope we all agree that our loved little worker queue service deserves a modern fresh-looking documentation oriented website.

I spent a little bit of time this weekend and would like to present for consideration a POC that is available at https://github.com/pmatseykanets/beanstalkd-docs

The intent was to combine the available documentation pieces and wiki pages under one umbrella and provide a modern, fresh and responsive look and feel.

It uses VuePress under the hood and is fully driven by markdown files and instantly deployable to GitHub pages or anywhere else for that matter.

It goes without saying that every bit of information as well as the way it structured in POC is for discussion and can be easily changed.

If you like the direction sources can live in a dedicated repository in `beanstalkd` organization i.e. `beanstalkd/docs`.

I'm willing to be the main maintainer for the site.

Let me know what you think.

**Comment by pmatseykanets** [comment]
Also things that are currently missing but would be great to have:

- a hero image
- color scheme
- a copy for most essential features
  - Simple
  - Fast
  - Reliable

**Comment by sergeyklay** [comment]
Yeah, we cam move all the wiki to the https://beanstalkd.github.io site.

**Comment by JensRantil** [comment]
Wow! This all sounds amazing. 💓 I'm currently out hiking but I'll have a look some time this week with thoughts and feedback.

**Comment by pmatseykanets** [comment]
@JensRantil Take look at the POC from your phone to see the mobile view while you're taking a rest during your hike 😅😅😅

**Comment by pmatseykanets** [comment]
@JensRantil Just checking in to see if you had an opportunity to take a look.

**Comment by DoogeJ** [comment]
Maybe mildly off-topic here, but is there still a list of available client libraries somewhere?

**Comment by stamster** [comment]
> Maybe mildly off-topic here, but is there still a list of available client libraries somewhere?

https://github.com/beanstalkd/beanstalkd/wiki/Client-Libraries

**Comment by pmatseykanets** [comment]
I didn't update the page since created the PoC but here it is https://pmatseykanets.github.io/beanstalkd-docs/resources/libraries.html

**Comment by pmatseykanets** [comment]
@ysmolsky @tomponline I know you have your hands full at the moment but would like to hear your thoughts as new maintainers at some point whether you feel like it's worth considering. Thanks.

/cc @kr

**Comment by ysmolsky** [comment]
@pmatseykanets I will be frank and maybe too blunt. I do not have capacity for this right now. Until I fix some critical bugs with segfaults (and one that is hard to replicate), I won't go into website business. I would like someone to takeover this completely. Would you be able to maintain and support such a website in future? I do not want to sound ungrateful, I appreciate any kind of help. I just have no idea when I will be able to pick this up. 

Some questions. Why VuePress? Where can we see the sources for this website? We might want to have release notes on the new website. Current version of beanstalkd has some kind of automation for publishing this changelog (News) via https://github.com/beanstalkd/beanstalkd/tree/master/pkg. Is it possible to replicate this in VuePress? Truth be told, I would like not to change the sources for website at this point since Jekyll is somewhat boring/static/stable.

I hope that you understand my position.

**Comment by tomponline** [comment]
Is it possible to do something like this but still use Jekyll or something similar that is supported by github pages generator, that way it would have minimal requirements on contributors to make updates and not require much maintenance.

**Comment by pmatseykanets** [comment]
@ysmolsky Both frank and blunt is fine with me :smile

> Would you be able to maintain and support such a website in future?

I think I'll be able to

> Why VuePress?

To be fair this type of question "Why X?" is not always easy to answer. There are multiple depends to it and of course there are usually alternatives. For example why did you choose Codecov over Coveralls, or Travis over CircleCI?
  
Now the description of the issue partially explains that VuePress is a modern **static** site generator specifically tailored for technical documentation sites. Are there alternatives? Or course there are.

> Where can we see the sources for this website?

https://github.com/pmatseykanets/beanstalkd-docs/

> Current version of beanstalkd has some kind of automation for publishing this changelog (News) via https://github.com/beanstalkd/beanstalkd/tree/master/pkg. Is it possible to replicate this in VuePress?

It's all markdown files underneath, so should be possible with the same type of scripting

> I would like not to change the sources for website at this point since Jekyll is somewhat boring/static/stable

It's the same kind of boring static site generator but focused on gorgeous documentation sites.
If you don't want to change it - fine with me. It's just I think such a beautiful piece of software (beanstalkd) which I myself used for years with great benefit deserves a great site. Thus the proposal.

@tomponline 

> Is it possible to do something like this but still use Jekyll 

Possibly, it's just VuePress is designed specially with tech doc sites in mind

> or something similar that is supported by github pages generator, 

VuePress is a static site generator and is **perfectly supported by GitHub pages**. In fact it's exactly how the prototype is hosted https://pmatseykanets.github.io/beanstalkd-docs/

> that way it would have minimal requirements on contributors to make updates and not require much maintenance.

It's already not more than than the requirements for Jekyll, probably less. It's just a set of markdown files. Take a look at the sources. Not sure what can be more minimal https://github.com/pmatseykanets/beanstalkd-docs/

Hope this make sense.

**Comment by tomponline** [comment]
Thanks, so does that mean the markdown files can just be edited and then pushed to github without the sites HTML files being generated and pushed, and then github pages will generate them? Cheers

**Comment by pmatseykanets** [comment]
Of course it’s just the build will be run by Travis or CircleCI and not GitHub itself. Although nowadays GitHub actions could be used if there is a free version of them for public repos. In any case there is nothing different from the contributor standpoint, and more explicit (less magic).

**Comment by pmatseykanets** [comment]
In fact, I just set it up to do just that with TravisCI. Any push to master publishes changes.

**Comment by tomponline** [comment]
@pmatseykanets apologies the long reply time, I've been rather busy recently, but I wanted to give you a proper response.

I know it's subjective, but I really like the layout of this new site.
It would be fantastic to have something like this for the beanstalkd web site.

But alas, I'm no expert in using npm, nodejs or VuePress. I did try the https://vuepress.vuejs.org/guide/getting-started.html#global-installation getting started guide, but fell at the first hurdle as `yarn` is not available in Ubuntu (`apt install yarn` installs an unrelated tool). Do I have to install nodejs first and then install yarn from node?

The JS ecosystem confuses me these days. :(

But that in itself is not a good reason to not proceed with this, as long as we have someone who does understand the ecosystem and is prepared to help maintain the site (keeping the framework version up to date etc). Hopefully you are that person @pmatseykanets :)

Whilst the current site is certainly looking dated, making amends is trivial. Even developing locally (at least on Ubuntu) is only an `apt install jekyll` and a `jekyll serve` away. I've also had some success with Hugo which is a static site generator using Go (although there are binaries available you can just download, and its available as `apt install hugo` in Ubuntu).

I'd like to get to a point where we have instructions for contributors (including myself and @ysmolsky) on how to setup the site locally for development that don't depend on knowing the JS ecosystem so that developing on the site is as easy as it is today.

@pmatseykanets what do you think are the next steps to move this forward?

Thanks
Tom
